tests/suites/main_test.function

```diff
@@ -239,6 +239,7 @@ int main( int argc, const char *argv[] )
 {
 #if defined(MBEDTLS_TEST_HOOKS) && defined (MBEDTLS_ERROR_C)
     mbedtls_test_hook_error_add = &mbedtls_test_err_add_check;
+    mbedtls_test_hook_value = &mbedtls_test_hook_value_check;
```

Comment:
I don't see why this should be guarded by MBEDTLS_ERROR_C, just MBEDTLS_TEST_HOOKS I'd say.